File: java/array-string/RemoveElement.java
class RemoveElement {
    public int removeElement(int[] nums, int val){
        int k = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
}

public class Main {
    public static void main(String[] args) {
        RemoveElement sol = new RemoveElement();

        int[] nums = {3, 2, 2, 3};
        int val = 3;

        int k = sol.removeElement(nums, val);

        System.out.println("k: " + k);
        System.out.println("modified nums: ");
        for (int i = 0; i < k; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}

File: javascript/array-string/RemoveElement.js
function removeElement(list, val){
    k = 0

    for(let i = 0; i < list.length ; i++){

        if (list[i] != val) {
            list[k] = list[i];
            k += 1;
        }
    }

    return k;
}

const array1 = [1, 3, 5, 7];

const result = removeElement(array1, 3);
console.log(`k: ${result}`)